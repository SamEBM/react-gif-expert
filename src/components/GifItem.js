import React from 'react';
import PropTypes from 'prop-types';

const GifItem = ({id, title, url}) => {
    return (
        <div className='card animate__animated animate__bounceIn'>
            <img alt={title} src={url}></img>
            <p>{title}</p>
        </div>
    )
}

GifItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}

export default GifItem