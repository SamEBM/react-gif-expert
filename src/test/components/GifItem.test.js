import GifItem from "../../components/GifItem";
import { shallow } from 'enzyme';

describe('Pruebas con Gif Item', () => { 
    const title = 'Titulo';
    const url = 'https://localhost/algo.jpg';
    let wrapper = shallow( <GifItem title={title} url={url} /> );

    // Esto se ejecuta cada vez que se ejecuta un test
    beforeEach( () => {
        wrapper = shallow( <GifItem title={title} url={url}/> );
    });

    test('Debe mostrar <GifItem /> correctamente', () => { 
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe tener un parrafo con Title', () => { 
        const parrafo = wrapper.find('p');
        expect(parrafo.text().trim()).toBe(title);
    });

    test('Debe tener un parrafo con Title', () => { 
        const img = wrapper.find('img');
        expect(img.prop('src').trim()).toBe(url);
        expect(img.prop('alt').trim()).toBe(title);
    });

    test('Debe tener clase animate__bounceIn', () => { 
        const divAnimation = wrapper.find('div');
        expect(divAnimation.prop('className').includes('animate__bounceIn')).toBe(true);
    });
})