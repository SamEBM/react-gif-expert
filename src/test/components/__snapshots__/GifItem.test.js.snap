// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con Gif Item Debe mostrar <GifItem /> correctamente 1`] = `
<div
  className="card animate__animated animate__bounceIn"
>
  <img
    alt="Titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
    Titulo
  </p>
</div>
`;
